### ISSUE TITLE
Subscription command issue?

### ISSUE THREAD

**Issue by darsenault** [tomarrell/miniqueue#4]
I love this queue -- go, simple, secure. Thank you. However I am having a terrible time getting subscriptions to work.

```
curl --insecure --cacert certs/ca.key --key certs/node.key --cert certs/node.crt -X POST https://localhost:9999/subscribe/foo                                                     
{"error":"error decoding command"}
```

Is it not possible to test subscriptions with curl? I publish just fine using the example in README. Thanks.

**Comment by darsenault** [comment]
Adding some detail:

Using the echo program from /examples and the mkcert setup. On macOS Catalina 10.15.7.

```
miniqueue % go run echo.go 
> foo
Published message foo to topic test_topic
Consumed message: foo
> bar
Published message bar to topic test_topic
Consumed message: bar
> hello there
Published message hello there to topic test_topic
Consumed message: hello there
> 
```

Getting **ERR failed decoding command error=EOF** error with the subscribe endpoint, but the data comes back to the client program. Welcome any suggestions/fixes.

```
4:14PM INF starting miniqueue port=:9999
4:14PM INF publishing to topic handler=publish request_id=c1397irrbdg6m98fropg topic=test_topic
4:14PM INF subscribing to topic handler=subscribe request_id=c1397irrbdg6m98froq0 topic=test_topic
4:14PM INF client disconnected while waiting for message cmd=ACK handler=subscribe request_id=c1397irrbdg6m98froq0 topic=test_topic
4:14PM ERR failed decoding command error=EOF handler=subscribe request_id=c1397irrbdg6m98froq0 topic=test_topic
4:15PM INF publishing to topic handler=publish request_id=c1397lrrbdg6m98fror0 topic=test_topic
4:15PM INF subscribing to topic handler=subscribe request_id=c1397lrrbdg6m98frorg topic=test_topic
4:15PM INF client disconnected while waiting for message cmd=ACK handler=subscribe request_id=c1397lrrbdg6m98frorg topic=test_topic
4:15PM ERR failed decoding command error=EOF handler=subscribe request_id=c1397lrrbdg6m98frorg topic=test_topic
4:15PM INF publishing to topic handler=publish request_id=c1397q3rbdg6m98frosg topic=test_topic
4:15PM INF subscribing to topic handler=subscribe request_id=c1397q3rbdg6m98frot0 topic=test_topic
4:15PM INF client disconnected while waiting for message cmd=ACK handler=subscribe request_id=c1397q3rbdg6m98frot0 topic=test_topic
4:15PM ERR failed decoding command error=EOF handler=subscribe request_id=c1397q3rbdg6m98frot0 topic=test_topic
```

**Comment by tomarrell** [comment]
G'day @darsenault, thanks a lot for issue.

Regarding setting up a subscriber, along with your subscribe POST you need to pass through the command "INIT" in order to specify creating a new subscriber. I understand this may seem counter-intuitive, but it is in order to leave some room to expand the subscription with group functionality in the future. Therefore, if you adjust your CURL to the following it should hopefully consume a single message. 

```
curl --insecure --cacert certs/ca.key --key certs/node.key --cert certs/node.crt -X POST https://localhost:9999/subscribe/foo -d "\"INIT\""
```

Keep in mind that CURL doesn't support keeping the connection open and continuously reading the body (or at least that I'm aware of), so I'd recommend you play around with the demos to get a better handle on how things work.

Regarding the errors you're seeing. Those are just due to client disconnects, I believe I fixed them in a recent release as they shouldn't be of `error` severity. However maybe I'm mistaken. I'll double check and sort them out if not.

Cheers!

**Comment by tomarrell** [comment]
Fixed with v0.8.0
